lib/amazonka-core/src/Amazonka/Endpoint.hs

```diff
@@ -95,3 +98,8 @@ defaultEndpoint Service {endpointPrefix = p} r = go (CI.mk p)
         }
 
     reg = toBS r
+
+    shouldMakeURLFromRegionSTS :: Bool
+    shouldMakeURLFromRegionSTS =
+      let mval = unsafePerformIO $ Environment.lookupEnv "SHOULD_MAKE_URL_FROM_REGION_STS"
```

Comment:
can we remove the usage of unsafePerformIO and make this value as input to defaultEndpoint function. 

If we can get away with not defining the env in this library it's even better.